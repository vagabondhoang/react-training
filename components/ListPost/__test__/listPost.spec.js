import React from 'react';
import Enzyme, { mount } from 'enzyme';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { ListPost } from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('ListPost component', () => {
  let component;
  let mockProps;

  const setUpComponent = (overrides = {}) => {
    mockProps = {
      posts: [
        {
          title: 'title1',
          user: {
            first: 'first1',
            last: 'last1',
          },
        },
      ],
      ...overrides,
    };

    component = mount(<ListPost {...mockProps} />);
  };

  beforeEach(() => {
    setUpComponent();
  });

  it('should ...', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
