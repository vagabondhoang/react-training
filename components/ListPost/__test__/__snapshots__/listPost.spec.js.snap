// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPost component should ... 1`] = `
.c0 {
  background: green;
  color: yellow;
}

<ListPost
  posts={
    Array [
      Object {
        "title": "title1",
        "user": Object {
          "first": "first1",
          "last": "last1",
        },
      },
    ]
  }
>
  <styled.li
    title="title1"
  >
    <li
      className="c0"
      title="title1"
    >
      title1 - first1 last1
    </li>
  </styled.li>
</ListPost>
`;
